go-controller/pkg/node/gateway_shared_intf.go

```diff
@@ -129,6 +129,10 @@ func (npw *nodePortWatcher) updateGatewayIPs(addressManager *addressManager) {
 // `add` parameter indicates if the flows should exist or be removed from the cache
 // `hasLocalHostNetworkEp` indicates if at least one host networked endpoint exists for this service which is local to this node.
 func (npw *nodePortWatcher) updateServiceFlowCache(service *kapi.Service, add, hasLocalHostNetworkEp bool) error {
+	if config.Gateway.Mode == config.GatewayModeLocal && config.Gateway.AllowNoUplink {
```

Comment:
What happens if there is an uplink even if `AllowNoUplink` is set?

It seems that `AllowNoUplink` is actually not equivalent to not having an uplink.

Reply:
Good question. I assume users shall only set this flag when they want to have a no-uplink gateway bridge. But I think we can check if there is the existence of the uplink port here.